crypto/provider_core.c

```diff
@@ -329,6 +382,46 @@ int ossl_provider_activate(OSSL_PROVIDER *prov)
     return 1;
 }
 
+int ossl_provider_activate(OSSL_PROVIDER *prov)
+{
+    if (provider_activate(prov)) {
```

Comment:
The comment above says `provider_activate` is the internal lockless version - but here we're just calling it anyway without obtaining a lock??

Reply:
The activation in itself doesn't need protection, it's already assumed that the reference count makes it legitimate to poke the activated flag.  The lock happens below, to set the store flag.

Reply:
In which case the comment above `provider_activate` is probably misleading and could do with rewording

Reply:
Okie

Reply:
Done